system-x/services/mail/src/main/java/software/tnb/mail/resource/openshift/OpenshiftMailServer.java

```diff
@@ -0,0 +1,167 @@
+                            .addAllToPorts(ports)
+                            .addAllToEnv(containerEnvironment().entrySet().stream().map(e -> new EnvVar(e.getKey(), e.getValue(), null))
+                                .collect(Collectors.toList()))
+                        .endContainer()
+                    .endSpec()
+                .endTemplate()
+                .addNewTrigger()
+                    .withType("ConfigChange")
+                .endTrigger()
+            .endSpec()
+            .build()
+        );
+
+        ServiceSpecBuilder serviceSpecBuilder = new ServiceSpecBuilder().addToSelector(OpenshiftConfiguration.openshiftDeploymentLabel(), name());
+
+        serviceSpecBuilder.addToPorts(new ServicePortBuilder()
+            .withName(name())
+            .withPort(smtpPort())
+            .withTargetPort(new IntOrString(smtpPort()))
+            .build());
+
+        LOG.debug("Creating service {}", name());
+        OpenshiftClient.get().services().createOrReplace(
+          new ServiceBuilder()
+            .editOrNewMetadata()
+                .withName(name())
+                .addToLabels(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+            .endMetadata()
+            .editOrNewSpecLike(serviceSpecBuilder.build())
+            .endSpec()
+            .build()
+        );
+        // @formatter:on
+    }
+
+    @Override
+    public void undeploy() {
+        LOG.info("Undeploying OpenShift Mail");
+        LOG.debug("Deleting service {}", name());
+        OpenshiftClient.get().services().withName(name()).delete();
+        LOG.debug("Deleting deploymentconfig {}", name());
+        OpenshiftClient.get().deploymentConfigs().withName(name()).delete();
+        OpenShiftWaiters.get(OpenshiftClient.get(), () -> false).areNoPodsPresent(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+            .timeout(120_000).waitFor();
+    }
+
+    @Override
+    public void openResources() {
+        LOG.debug("Creating port-forward to {} for port {}", name(), smtpPort());
+        portForward = OpenshiftClient.get().services().withName(name()).portForward(smtpPort(), smtpPort());
+        LOG.debug("Creating port-forward to {} for port {}", name(), httpPort());
+        httpPortForward = OpenshiftClient.get().services().withName(name()).portForward(httpPort(), httpPort());
+    }
+
+    @Override
+    public boolean isReady() {
+        List<Pod> pods = OpenshiftClient.get().getLabeledPods(OpenshiftConfiguration.openshiftDeploymentLabel(), name());
+        if (ResourceFunctions.areExactlyNPodsReady(1).apply(pods)) {
+            return OpenshiftClient.get().getLogs(pods.get(0)).contains("Transition to primary complete; database writes are now permitted");
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isDeployed() {
+        return OpenshiftClient.get().getLabeledPods(OpenshiftConfiguration.openshiftDeploymentLabel(), name()).size() != 0
+            && isReady();
```

Comment:
remove `isReady` (I know it's copy-paste, I need to remove it from other openshift resources as well)